### PR TITLE
chore: Use upstream cypress-testing-library again

### LINKED ISSUE #24529: Update cypress-testing-library dependency back to upstream after they update to queries

### PR DESCRIPTION
**Pull request by BlueWinds**
- Closes #24529 

### User facing changelog
No user facing changes.

### Additional details
During the rollout of Cypress 12, we used a fork of `@testing-library/cypress`. They've now merged in our changes, so it's no longer necessary. This PR moves Cypress back to using the upstream version rather than our fork.

### PR Tasks

- [n/a] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by emilyrohrbough — APPROVED** [review]

File: packages/example/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   "devDependencies": {
     "chai": "3.5.0",
     "cross-env": "6.0.3",
-    "cypress-example-kitchensink": "https://github.com/cypress-io/cypress-example-kitchensink.git#bd93f68568bc0a412b2c4dd2ddbc3b7cb89e5799",
```
Comment (emilyrohrbough) [inline packages/example/package.json]: Why was this bumped?

Comment (emilyrohrbough) [reply packages/example/package.json]: We should add in summary/pr title if we need to bump.

Comment (BlueWinds) [reply packages/example/package.json]: I committed this change to the wrong branch, doesn't need changing. Good catch, will undo.

Comment (BlueWinds) [reply packages/example/package.json]: Had meant to put this change in a different PR.

**Review by emilyrohrbough — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Comment by BlueWinds** [comment]
The two failing tests are unrelated and flaky, going to merge this in.

**Pull request merged**